smallrye-reactive-messaging-rabbitmq/src/main/java/io/smallrye/reactive/messaging/rabbitmq/RabbitMQConnector.java

```diff
@@ -261,7 +261,7 @@ public void terminate(
             outgoing.terminate();
         }
 
-        clients.forEach((channel, rabbitMQClient) -> rabbitMQClient.stopAndForget());
+        clients.forEach((channel, rabbitMQClient) -> rabbitMQClient.stopAndAwait());
```

Comment:
Are you sure we are never called on the event loop? 

Maybe we need to be more defensive.

Reply:
For sure CDI events are called on worker threads. 

We used to await before actually. We had a CI issue which blocked on await. But I think it was because we didn't close the consumers.